### DISCUSSION TITLE
regex manager: support "replace" in templates

### DISCUSSION THREAD

**Discussion by dmivankov** [renovatebot/renovate#16332] [Request Help]
### What would you like Renovate to be able to do?

Regex manager should be able to handle dependencies like `https://maven.com/com/example/foobar/1.0/foobar-1.0.jar` and map that to `depName=com.example:foobar, currentValue=1.0`

The issue is that there seems to be no replace or regex replace handlebars template helper available in renovatebot.
Without replace depName can be extracted as `com/example/foobar` or as `com/example:foobar` while it should be `com.example:foobar`, the issue is that `com/example` part may have any number of path tokens, workaround is to implement say 1-10 matchers one for each path depth

### If you have any ideas on how this should be implemented, please tell us here.

Something like following may be supported
```json
{
   "matchStrings": ["https:\/\/maven\\.com\/(?<path>.*?)\/(?<last>[^/]*?)\/(?<currentValue>[^/]*?)\/[^/]*\\.jar"],
   "datasourceTemplate": "maven",
   "depNameTemplate": "{{#replace \"/\" \".\"}}{{{path}}}{{/replace}}:{{{last}}}"
}
```

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
We have a replace helper for handlebars already. Does that help?

**Comment by dmivankov** [comment]
what's the syntax?
```
      "depNameTemplate": "{{#replace \"/\" \".\"}}{{{path}}}{{/replace}}:{{{last}}}"
```
gets 
>        "error": {
>         "validationSource": "renovate.json",
>         "validationError": "The renovate configuration file contains some invalid settings",
>         "validationMessage": "Invalid template in config path: regexManagers[0].depNameTemplate",
>         "message": "config-validation",
while
```
      "depNameTemplate": "{{{path}}}:{{{last}}}"
```
works, producing "com/example:foobar"

**Comment by viceice** [comment]
https://docs.renovatebot.com/templates/#replace

**Comment by viceice** [comment]
@rarkins should be converted to discussion

**Comment by dmivankov** [comment]
```
      "depNameTemplate": "{{{replace '/' '.' path}}}:{{{last}}}"
```
this works!
Would be great to add a link from https://docs.renovatebot.com/modules/manager/regex/ to available templates

**Reply by HonkingGoose** [reply]
See PR:

- #16340
